### PR TITLE
docs: CHARTS-8818 add rendering spec documentation

### PR DESCRIPTION
**Pull request by mongo-J**
## Summary
This PR adds:
- Documentation for Embedded Chart Rendering Specification 
- Link to the Rendering spec doc from the README file

### CODE REVIEW

File: docs/rendering-spec.md [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+| labelOverride | string                                  | Overriding the label of the axis.                                                                                                                                                                                                                                                                                                                             |
+| labelAngle    | 'vertial' \| 'horizontal' \| 'diagonal' | Applicable for the following chart types only: All Column charts, 100% Stacked Area, Discrete Area, Discrete Line, Grouped Combo, Stacked Combo, Heatmap.                                                                                                                                                                                                     |
```
Comment (khanguslee) [inline docs/rendering-spec.md:41]:
```suggestion
| labelAngle    | 'vertical' \| 'horizontal' \| 'diagonal' | Applicable for the following chart types only: All Column charts, 100% Stacked Area, Discrete Area, Discrete Line, Grouped Combo, Stacked Combo, Heatmap.                                                                                                                                                                                                     |
```

File: docs/rendering-spec.md [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+| legendPosition                  | 'right' \| 'bottom' \| "top"                                                                                                                                                                                                                                            | Applicable to all chart types that support a legend.                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (khanguslee) [inline docs/rendering-spec.md:54]:
```suggestion
| legendPosition                  | 'right' \| 'bottom' \| 'top'                                                                                                                                                                                                                                            | Applicable to all chart types that support a legend.                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```

File: docs/rendering-spec.md [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+| colorPalette                    | object \| false                                                                                                                                                                                                                                                         | Applicable for most chart types depending on the color palette type - discrete or continuous. Use `false` to unset the custom colorPalette that was set from the chartBuilder (if any).                                                                                                                                                                                                                                                                                                    |
```
Comment (khanguslee) [inline docs/rendering-spec.md:61]:
```suggestion
| colorPalette                    | object \| false                                                                                                                                                                                                                                                         | Applicable for most chart types depending on the color palette type - discrete or continuous. Use `false` to unset the custom colorPalette that was set from the chart builder (if any).                                                                                                                                                                                                                                                                                                    |
```

File: docs/rendering-spec.md [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+| colorPalette.channels           | Array of Charts channel names that will receive a custom color                                                                                                                                                                                                          | Applicable for all chart types that can have multi-series with multiple fields in the same channel:                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (khanguslee) [inline docs/rendering-spec.md:66]:
```suggestion
| colorPalette.channels           | Array of Charts channel names that will receive a custom color                                                                                                                                                                                                          | Applicable for all chart types that can have multi-series with multiple fields in the same channel.                                                                                                                                                                                                                                                                                                                                                                                        |
```

File: docs/rendering-spec.md [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+
+### Discrete rules
+
+Applicable for Table and Number charts
+|Customization|Type|Description|
+|-|-|-|
+|type|'discrete'||
+|backgroundColor|string (rgb colors, hex colors, word colors) 'transparent' (this means that the color will be taken from a previous rule, if any)|Examples:<br/> 'rgb(240, 209, 117)' , 'limegreen', '#04910e'|
+|textColor|string (rgb colors, hex colors, word colors)
+'auto' (no color)||
```
Comment (khanguslee) [inline docs/rendering-spec.md:186]: This has an extra cell

File: docs/rendering-spec.md [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+#### Example for conditions using a different type of operator
+
+NumericOperator for Table charts (similar for NumericBinnedOperator):
+
+```javascript
+{
+operator: 'NUMBER_GREATER_THAN',
+targetType: 'CHANNEL',
+targetChannel: 'value_series_0',
+value: 190,
```
Comment (khanguslee) [inline docs/rendering-spec.md:269]: Add spacing

**Review by khanguslee — CHANGES_REQUESTED** [review]
Nice work @mongo-J and @kristinamongo on the documentation! Just some minor mistakes to be fixed and then we can merge it in.

File: docs/rendering-spec.md (hunk 1 above)
Comment (mongo-J) [reply docs/rendering-spec.md]: fixed all the typos and format issues here: https://github.com/mongodb-js/charts-embed-sdk/pull/86/commits/731549b69f634969e5f2a52feb0e501cc8ba132a

File: docs/rendering-spec.md [hunk 7]
```diff
@@ -0,0 +1,523 @@
...
+| colorPalette.values             | Array with color values that will be used to customize the colors for single series charts or “channels” and “marks” of multi-series charts                                                                                                                             |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
+
+### Color palette customizations
+
+For chart with continuous colors:
```
Comment (lilyh4) [inline docs/rendering-spec.md:73]:
```suggestion
For charts with continuous colors:
```

File: docs/rendering-spec.md [hunk 8]
```diff
@@ -0,0 +1,523 @@
...
+    colorPalette: {
+      type: "continuous",
+      color: "rainbow",
+      reverseColor: true,
+    },
+  },
+});
+```
+
+For chart with single series:
```
Comment (lilyh4) [inline docs/rendering-spec.md:88]:
```suggestion
For charts with single series:
```

File: docs/rendering-spec.md [hunk 9]
```diff
@@ -0,0 +1,523 @@
...
+    colorPalette: {
+      type: "discrete",
+      channels: ["x", "x_series_0", "x_series_1"],
+      value: ["aqua", "rgb(49, 74, 63)", "olive"],
+    },
+  },
+});
+```
+
+For Candlestick chart:
```
Comment (lilyh4) [inline docs/rendering-spec.md:132]:
```suggestion
For Candlestick charts:
```

File: docs/rendering-spec.md [hunk 10]
```diff
@@ -0,0 +1,523 @@
...
+
+### Other methods usage
+
+You can get the rendering spec customizations that you have set through the SDK by calling the getter method:
+
+```javascript
+const customizations = chart.getRenderingSpecOverride();
+```
+
+If you are wondering what channels or axes does you embedding chart have, you can call the following methods:
```
Comment (lilyh4) [inline docs/rendering-spec.md:431]:
```suggestion
If you are wondering what channels or axes your embedded chart has, you can call the following methods:
```

File: docs/rendering-spec.md [hunk 11]
```diff
@@ -0,0 +1,523 @@
...
+| Table               | **Channels**: <br/>- group, value, dynamicColumns<br/><br/>**Axes**:<br/>N/A                   | **Channel customizations**:<br/>- labelOverride, numberMultiplier, numberDecimals, numberSuffix, numberThousandsSeparators, numberPrefix, wrapText<br/><br/>**Option customizations**:<br/>- textSize, totalsColumn, totalsRow<br/><br/>**ConditionalFormatting**                                                                                                                    |
+
+Note that the channels for multi-series with multiple fields in the same channel are dynamic and will look like the following: x_series_0, x_series_1, x_series_2, etc. The following channels can have mutliseries: “x”, “y”, “y_lines”, “group”, “value”.
```
Comment (lilyh4) [inline docs/rendering-spec.md:503]:
```suggestion
Note that the channels for multi-series with multiple fields in the same channel are dynamic and will look like the following: x_series_0, x_series_1, x_series_2, etc. The following channels can have multi-series: “x”, “y”, “y_lines”, “group”, “value”.
```

File: docs/rendering-spec.md [hunk 12]
```diff
@@ -0,0 +1,523 @@
...
+      labelOverride: "new label",
+    },
+    x_series_0: {
+      labelOverride: "new series label",
+    },
+  },
+});
+```
+
+If you have applied a customization but do not see the result applied in the embedded chart, nor there is any error in the browser console - this means that the rendering spec has been parsed successfully but is incorrect. In such cases, we recommend that you check all single keys you are setting in the docs one by one and also check their applicability per the chart type.
```
Comment (lilyh4) [inline docs/rendering-spec.md:523]:
```suggestion
If you have applied a customization but do not see the result applied in the embedded chart, nor is there any error in the browser console, this means that the rendering spec has been parsed successfully but is incorrect. In such cases, we recommend that you check all single keys you are setting in the docs one by one and also check their applicability per the chart type.
```

**Review by lilyh4 — APPROVED** [review]
These docs look great! LGTM with a couple of typos and grammar nits.

File: docs/rendering-spec.md (hunk 7 above)
Comment (mongo-J) [reply docs/rendering-spec.md]: Thanks Lily, I have fixed these here: https://github.com/mongodb-js/charts-embed-sdk/pull/86/commits

**Review by khanguslee — APPROVED** [review]
LGMT 🥳

**Pull request merged**